Realm/Realm/Handles/SharedRealmHandle.cs

```diff
@@ -48,6 +48,8 @@ internal class SharedRealmHandle : StandaloneHandle
         // it is important that children always have a reference path to their root for this to work
         private bool _noMoreUserThread;
 
+        private uint? _asyncTransactionHandle;
```

Comment:
I think we should remove this field because even with cancellations implemented, I don't suppose we'll ever store the handle as a field-level variable. For now, we can disregard the returned values from begin/commit async and start using them in your next PR.